### PR TITLE
Adds text for Challenge 1 - Azure Functions as Change Feed triggers

### PR DESCRIPTION
**Pull request by codepossible**
## Primary Changes 
- Adds description for Challenge 1
- Adds new solution design image - circling the challenge area.

### CODE REVIEW

File: docs/challenge-001.md [hunk 1]
```diff
@@ -1,9 +1,43 @@
...
+
+In this challenge we will begin that processing by building Change Feed Listeners using Azure Functions.
+
+## Guidance
+
+As part of this challenge we are focusing on the following red circled area of the design.
+
+![Focus for Challenge 1](./images/OpenHack_ProcessingFlow_Challenge01Focus.png)
+
+To accomplish this challenge, you will have to connect to your Azure Account and create a new Azure CosmosDB Account and database called - ```inventory``` using the **SQL API**.
```
Comment (davidkpiano) [inline docs/challenge-001.md]:
```suggestion
To accomplish this challenge, you will have to connect to your Azure Account and create a new Azure CosmosDB Account and database called - `inventory` using the **SQL API**.
```

File: docs/challenge-001.md [hunk 2]
```diff
@@ -1,9 +1,43 @@
...
+
+In the CosmosDB database, create three containers - ```shipments``` (partition key: /id) and ```onhand``` (partition key: /id) and mdc  (partition key: /id).
```
Comment (davidkpiano) [inline docs/challenge-001.md]:
```suggestion
In the CosmosDB database, create three containers - `shipments` (partition key: `/id`) and `onHand` (partition key: `/id`) and `mdc` (partition key: `/id`).
```

File: docs/challenge-001.md [hunk 3]
```diff
@@ -1,9 +1,43 @@
...
+
+For the change feed listeners, create two Azure Functions with CosmosDB as the trigger. When triggered, the Azure function list out the number of changed document in the change feed and id of the first document.
+
+Insert a document in ```onhand``` and ```shipment``` collection using Azure Storage Explorer and validate the Azure Function as been triggered.
```
Comment (davidkpiano) [inline docs/challenge-001.md]:
```suggestion
Insert a document in `onHand` and `shipment` collection using Azure Storage Explorer and validate the Azure Function as been triggered.
```

File: docs/challenge-001.md [hunk 4]
```diff
@@ -1,9 +1,43 @@
...
+- [C# Azure Functions - GitHub Samples](https://github.com/Azure-Samples/functions-docs-csharp)
+
 ## Challenge Completion Criteria
+
+To successfully, complete this challenge, we should have:
+
+- [ ] Created a Azure CosmosDB database named - ```inventory```.
+- [ ] The Database should have three containers - ```onhand```, ```shipments``` and ```mdc```.
+- [ ] There should be two change feed listeners configured to listen change feed for the ```onhand``` and ```shipments``` containers.
+- [ ] The respective Azure functions should be triggered as documents are added to ```onhand``` and ```shipments``` containers.
```
Comment (davidkpiano) [inline docs/challenge-001.md]:
```suggestion
- [ ] Created a Azure CosmosDB database named - `inventory.
- [ ] The Database should have three containers - `onHand`, `shipments` and `mdc`.
- [ ] There should be two change feed listeners configured to listen change feed for the `onHand` and `shipments` containers.
- [ ] The respective Azure functions should be triggered as documents are added to `onHand` and `shipments` containers.
```

File: docs/challenge-001.md [hunk 5]
```diff
@@ -1,9 +1,43 @@
...
+
+As part of this challenge we are focusing on the following red circled area of the design.
+
+![Focus for Challenge 1](./images/OpenHack_ProcessingFlow_Challenge01Focus.png)
+
+To accomplish this challenge, you will have to connect to your Azure Account and create a new Azure CosmosDB Account and database called - `inventory` using the **SQL API**.
+
+In the CosmosDB database, create three containers - `shipments` (partition key: `/id`) and `onHand` (partition key: `/id`) and `mdc` (partition key: `/id`).
+
+For the change feed listeners, create two Azure Functions with CosmosDB as the trigger. When triggered, the Azure function list out the number of changed document in the change feed and id of the first document.
```
Comment (gavination) [inline docs/challenge-001.md]: Perhaps a section with definitions would be useful here? I didn’t know what a change feed was supposed to mean when I started, for example.

File: docs/challenge-001.md [hunk 6]
```diff
@@ -1,9 +1,43 @@
...
+
+## Guidance
+
+As part of this challenge we are focusing on the following red circled area of the design.
+
+![Focus for Challenge 1](./images/OpenHack_ProcessingFlow_Challenge01Focus.png)
+
+To accomplish this challenge, you will have to connect to your Azure Account and create a new Azure CosmosDB Account and database called - `inventory` using the **SQL API**.
+
+In the CosmosDB database, create three containers - `shipments` (partition key: `/id`) and `onHand` (partition key: `/id`) and `mdc` (partition key: `/id`).
```
Comment (gavination) [inline docs/challenge-001.md]: Can you talk about the purpose of the mdc container? Onhand and update are mentioned before but this is the first the reader learns of the mdc

File: docs/challenge-001.md (hunk 5 above)
Comment (codepossible) [reply docs/challenge-001.md]: Excellent point

File: docs/challenge-001.md (hunk 6 above)
Comment (codepossible) [reply docs/challenge-001.md]: Good Point, I will add that here. There is a solution design document covers some of that. But a reiterating that will help.

Comment (brandonh-msft) [reply docs/challenge-001.md]: I'd just change this to a bulleted list:
> create the following containers:
> - `shipments` (partition key `/id`) 
> ...

**Review by brandonh-msft — COMMENTED** [review]
minor suggestion on container list

File: docs/challenge-001.md (hunk 6 above)
Comment (codepossible) [reply docs/challenge-001.md]: Got it. Thanks @brandonh-msft

Comment (brandonh-msft) [reply docs/challenge-001.md]: could even do it as an MD table:

name | partition key
--- | ---
shipments | /id

File: docs/challenge-001.md [hunk 7]
```diff
@@ -1,9 +1,43 @@
...
+
+For the change feed listeners, create two Azure Functions with CosmosDB as the trigger. When triggered, the Azure function list out the number of changed document in the change feed and id of the first document.
+
+Insert a document in `onHand` and `shipment` collection using Azure Storage Explorer and validate the Azure Function as been triggered.
```
Comment (jcocchi) [inline docs/challenge-001.md]: Should we have them use the data generator at this point to create and submit documents of the right shape?

**Review by brandonh-msft — APPROVED** [review]

**Comment by codepossible** [comment]
Updated the Challenge 1 document to encourage exploration, based on discussion with @jcocchi, @davidkpiano  and @ianphil.

The happy middle was to provide specific guidance to technology to be user, input and output contacts (define schema for documents in Cosmos DB) but let the participants decide on the exchange of messages between the internal components.

Please review and comment/approve.

File: docs/challenge-001.md [hunk 8]
```diff
@@ -1,9 +1,58 @@
...
+- [Azure CosmosDB - Change Feeds](https://docs.microsoft.com/en-us/azure/cosmos-db/change-feed)
+- [Azure Functions](https://azure.microsoft.com/en-us/services/functions/)
+- [Azure Functions Triggers and Bindings](https://docs.microsoft.com/en-us/azure/azure-functions/functions-triggers-bindings)
+- [Azure Functions - CosmosDB Change Feed Trigger](https://docs.microsoft.com/en-us/azure/azure-functions/functions-create-cosmos-db-triggered-function)
+- [C# Azure Functions - GitHub Samples](https://github.com/Azure-Samples/functions-docs-csharp)
+
+### Terminology
+
+- **Azure CosmosDB Change Feed**: A change feed in AzureDB CosmosDB is a list of documents that have been updated or inserted.
+- **Partitioned Collection** : A terms used previous for CosmosDB containers
```
Comment (jcocchi) [inline docs/challenge-001.md]: The term `Partitioned Collection`  isn't used anywhere in this document so this definition might be more confusing than it is helpful. Both of the terms `collection` and `container` are valid for the SQL API, I would leave out this line all together

File: docs/challenge-001.md [hunk 9]
```diff
@@ -1,9 +1,58 @@
...
+
 ## Challenge Completion Criteria
+
+To successfully, complete this challenge, we should have:
+
+- [ ] Created a Azure CosmosDB database named - `inventory.
```
Comment (jcocchi) [inline docs/challenge-001.md]:
```suggestion
- [ ] Created a Azure CosmosDB database named - `inventory`.
```

File: docs/challenge-001.md [hunk 10]
```diff
@@ -1,9 +1,58 @@
...
+- [ ] The Database should have three containers - `onHand`, `shipments` and `mdc`.
+- [ ] There should a way to detect inserts and updates to the `onHand` and `shipments` containers.
+- [ ] The processing of the events should show the number of changed documents and id of the first document for each collection.
+
+## Next Steps
+
+Now we have CosmosDB change feed rolling into Azure Functions, let's do something more **exciting** with in our next challenge.
```
Comment (jcocchi) [inline docs/challenge-001.md]:
```suggestion
Now we have CosmosDB change feed rolling into Azure Functions, let's do something more **exciting** with the data in our next challenge.
```

**Review by jcocchi — APPROVED** [review]
Added a couple comments, but I really love the changes you made! Reads much more like an Open Hack now

File: docs/challenge-001.md (hunk 8 above)
Comment (codepossible) [reply docs/challenge-001.md]: Ok, fair enough. I will remove this. One term terminology section seems little empty, so I might add something else, when I can think of it.

File: docs/challenge-001.md (hunk 7 above)
Comment (codepossible) [reply docs/challenge-001.md]: We can include that in a story for "moving the data generator" as one of the acceptance criteria to update the documentation .

**Pull request merged**